snappy-tools/sbin/collect-debug-artifacts.sh

```diff
@@ -0,0 +1,230 @@
+      ${INTERVAL_BETWEEN_DUMPS} ${GET_EVERYTHING} ${collector_host} ${out_dir} ${VERBOSE}"
+}
+
+function collect_on_remote {
+  data_dir=$1
+  num_stack_dumps=$2
+  int_stack_dumps=$3
+  get_all=$4
+  collector_host=$5
+  collector_dir=$6
+  verbose=$7
+
+  # first get the pid. The latest log file with the header will have the pid
+  host=`hostname`
+  if [ ! -d $data_dir ]; then
+    echo "${data_dir} not found on host: ${host}"
+  fi
+
+  cd $data_dir
+  files=()
+  last_restart_log=""
+  for l in $( ls -t *.log )
+  do
+    copyright_headers=`grep 'Copyright (C)' ${l}`
+    if [ ! -z "$copyright_headers" ]; then
+      # also check for the pid line and get the pid
+      proc_id=`sed -n 's/.*Process ID: \([0-9]\+\)$/\1/p' ${l}`
+      if [ "${verbose}" = "1" ]; then
+        echo "Addiing file ${l} to the array"
+      fi
+      files+=($l)
+      last_restart_log=$l
+      break
+    fi
+  done
+
+  if [ -z "${proc_id}" ]; then
+    echo "No valid process id could be obtained from logs on ${host}"
+    exit 1
+  fi
+
+  # get the stack dumps first
+  dump_num=1
+  for i in {1..${num_stack_dumps}}
+  do
+    dump_num=`expr ${dump_num} + 1`
+    kill -URG $proc_id
+    sleep $int_stack_dumps
+  done
+
+  if [ "${get_all}" = "1" ]; then
+    if [ "${verbose}" = "1" ]; then
+      rsync -avrz $data_dir "${collector_host}:${collector_dir}/"
+    else
+      rsync -avrzq $data_dir "${collector_host}:${collector_dir}/"
+    fi
+  else
+    # get the latest log and the latest with the copyright headers
+    for l in $( ls -t *.log )
+    do
+      files+=($l)
+      if [ "${l}" = "${last_restart_log}" ]; then
+        break
+      fi
+    done
+    # get the latest stats file
+    for l in $( ls -t *.gfs 2> /dev/null )
+    do
+      if [ "${verbose}" = "1" ]; then
+        echo "Addiing file ${l} to the array"
+      fi
+      files+=($l)
+    done
+  fi
+  if [ "${verbose}" = "1" ]; then
+    echo "FILES=${files[@]} will be rsynced to ${collector_host}:$collector_dir}"
+  fi
+  if [ "${verbose}" = "1" ]; then
+    rsync -avz ${files[@]} "${collector_host}:${collector_dir}"
+  else
+    rsync -avzq ${files[@]} "${collector_host}:${collector_dir}"
+  fi
+}
+
+check_configs
+# Assuming each line in the members info file has the following format
+# host pid cwd
+
+all_pids=()
+# Make output directory
+rm -rf debug_data 2> /dev/null
+mkdir debug_data
+
+serv_num=1
+while  read -r line || [[ -n "$line" ]]; do
+  if [ "${VERBOSE}" = "1" ]; then
+    echo "Line read from file: $line"
+  fi
+
+  read host cwd <<< $line
+  if [ "${VERBOSE}" = "1" ]; then
+    echo "host: $host pid: $pid and cwd: $cwd"
+  fi
+
+  collect_data $host $cwd $serv_num &
+  serv_num=`expr $serv_num + 1`
+  all_pids+=($!)
+done < $MEMBERS_INFO_FILE
+
+for p in "${all_pids[@]}"
+do
+  if [ "${VERBOSE}" = "1" ]; then
+    echo "Waiting for pid ${p}"
+  fi
+  wait $p 2> /dev/null
+done
+
+# make zipped tar ball
+tar -zcf debug_data.tar.gz debug_data
```

Comment:
can we have an option to create individual zip per server because logs can be quite big at times.